### PR TITLE
done basic game

### PR DESCRIPTION
**Pull request by jcarrera94**
link: https://jcarrera94.github.io/2020-04-15-Conways-Game-of-Life/

### CODE REVIEW

**Review by Scott47 — COMMENTED** [review]
This is first place in my review.

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    });
+
+    setTimeout(runSimulation, speed);
+  }, [grid]);
+
+
+
+  return (
+    <div className="App">
+      <h1>Welcome to Conway's way of life</h1>
```
Comment (jcarrera94) [inline src/App.js:70]: I accidentally put Conway's way of life instead of Game of life 🤣

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  [-1, 0]
+];
+
+function App() {
+
+  const [grid, setGrid] = useState();
+  const [cols, setCols] = useState();
+  const [rows, setRows] = useState();
+  const [running, setRunning] = useState(false);
+  const [speed, setSpeed] = useState(1000)
```
Comment (SirPotatoIV) [inline src/App.js:23]: You may have planned on doing this in the future, but at some point it may be better to switch to useContext to organize all of your state in one spot.

File: src/App.js (hunk 1 above)
Comment (srsexton94) [reply src/App.js:70]: giphy

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,25 @@
 node_modules/
```
Comment (srsexton94) [inline .gitignore:1]: Hmm, I'm just unfamiliar - why was this included in a git push if it's a hidden file?

File: src/components/controls.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      <label>
+        Rows:
+        <input onChange={({target: {value}}) => setRows(value)} type="number" name="rows" />
+      </label>
+      <label>
+        Columns:
+        <input onChange={({target: {value}}) => setCols(value)} type="number" name="cols" />
+      </label>
+      <button onClick={() => setGrid(rows, cols)}>Set Grid</button>
+    </div>
```
Comment (SirPotatoIV) [inline src/components/controls.js:19]: The simplicity here is awesome.

File: src/App.js [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+              }
+            });
+
+            if (neighbors < 2 || neighbors > 3) {
+              gridCopy[i][k] = 0;
+            } else if (grid[i][k] === 0 && neighbors === 3) {
+              gridCopy[i][k] = 1;
+            }
+          }
+        }
```
Comment (SirPotatoIV) [inline src/App.js:59]: This is crazy that you fit all of your logic and grid building all in this file. It does make it hard to follow though. Maybe this time constraints this was needed, but it would be nice if this more broken out and defined. Maybe even in separate files and sub-functions. Super clever though.

**Review by srsexton94 — COMMENTED** [review]
Great job Juan!  Very impressive - using React was a great idea & you certainly wield it well - if you had more time is there anything else you'd want to add?

File: src/App.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
+import React, { useState, useCallback, useRef } from 'react';
```
Comment (srsexton94) [inline src/App.js:1]: This is structured really well!  I think some commentary throughout might've helped with legibility for code review - but still certainly readable!

**Comment by SirPotatoIV** [comment]
This is number 1 in my opinion
